eclair-core/src/main/scala/fr/acinq/eclair/wire/TlvCodecs.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.wire
+
+import fr.acinq.eclair.wire.LightningMessageCodecs._
+import scodec.codecs._
+import scodec.Codec
+
+/**
+  * Created by t-bast on 20/06/2019.
+  */
+
+object TlvCodecs {
+
+  val genericTlv: Codec[GenericTlv] = (("type" | varInt) :: variableSizeBytesLong(varInt, bytes)).as[GenericTlv]
+
+  def tlvFallback(codec: Codec[Tlv]): Codec[Tlv] = discriminatorFallback(genericTlv, codec).xmap(_ match {
```

Comment:
Nice find, `discriminatorFallback` seems like the right tool for this job.

Nit: should you type the codec to `DiscriminatorCodec`? Probably not, otherwise the right codec would have had this type in the definition of `discriminatorFallback`.

Reply:
> Nice find, discriminatorFallback seems like the right tool for this job.

Full credit goes to @sstone who found it :)

I didn't notice the `DiscriminatorCodec` provided by `scodec`, I'll try it out to see if it makes sense.

Reply:
From what I understand the `DiscriminatorCodec` type is mainly used internally to `scodec` when we use functions like `discriminatedBy`. I think what we have works well, I don't see a meaningful way of changing that to be more idiomatic.